File: examples/hello/FStar_Seq.ml
module Seq = BatSeq
type 'a seq = 'a Seq.t

let length s = Seq.length s
let index s i = Seq.at s i

let create n x = Seq.make n x

let createEmpty = Seq.nil

let upd s n x =
  let prefix = Seq.take n s in
  let suffix = Seq.drop (n + 1) s in
  Seq.append prefix (Seq.cons x suffix)

let append s1 s2 =
  Seq.append s1 s2

let slice s i j =
  let sj = Seq.take s j in
  Seq.drop i sj

File: ulib/js/FStar_Seq.js
/* @flow */

import * as FStar_Array from "./FStar_Array";

export type seq<T> = T[];

export let create = FStar_Array.create;
export let append = FStar_Array.append;
export let length = FStar_Array.length;
export let index  = FStar_Array.index;
export let eq = FStar_Array.eq; 
